src/commands/commit/index.js

```diff
@@ -3,21 +3,28 @@ import inquirer from 'inquirer'
 
 import getEmojis from '../../utils/getEmojis'
 import prompts from './prompts'
-import withHook, { registerHookInterruptionHandler } from './withHook'
+import withHook, {
+  registerHookInterruptionHandler,
+  cancelIfRebasing
+} from './withHook'
 import withClient from './withClient'
 
 const commit = (mode: 'client' | 'hook') => {
-  if (mode === 'hook') registerHookInterruptionHandler()
+  if (mode === 'hook') {
+    registerHookInterruptionHandler()
+  }
 
-  return getEmojis()
-    .then((gitmojis) => prompts(gitmojis))
-    .then((questions) => {
-      inquirer.prompt(questions).then((answers) => {
-        if (mode === 'hook') return withHook(answers)
+  return cancelIfRebasing(mode).then(() =>
```

Comment:
There's something I think we can improve here.

If we're already making a conditional for the `mode` inside of this function maybe we can reuse it and execute the `cancelIfRebasing` function here

Reply:
Sadly, it's not possible to use a function with await in if(mode) since then:
- We need to amke commit an async func
- Then in 'client' mode, we have the 'hang' that you observed in hook mode (the cli doesn't complete until ctrl-C)